components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/utxo/impl/UtxoPersistenceServiceImpl.kt

```diff
@@ -335,35 +350,150 @@ class UtxoPersistenceServiceImpl(
+                ).toByteArray()
+
+                val metadataHash = sandboxDigestService.hash(
+                    metadataBytes,
+                    DigestAlgorithmName.SHA2_256
+                ).toString()
+
+                repository.persistTransactionMetadata(
+                    em,
+                    metadataHash,
+                    metadataBytes,
+                    requireNotNull(metadata.getMembershipGroupParametersHash()) {
+                        "Metadata without membership group parameters hash"
+                    },
+                    requireNotNull(metadata.getCpiMetadata()) {
+                        "Metadata without CPI metadata"
+                    }.fileChecksum
+                )
+
+                repository.persistTransaction(
+                    em,
+                    filteredTransaction.id.toString(),
+                    filteredTransaction.privacySalt.bytes,
+                    account,
+                    utcClock.instant(),
+                    TransactionStatus.VERIFIED,
+                    metadataHash,
+                    isFiltered = true
+                )
+
+                filteredTransaction.filteredComponentGroups.forEach { (groupIndex, groupData) ->
+
+                    // Persist the merkle proof for each component group
+                    val persistedMerkleProofId = repository.persistMerkleProof(
+                        em,
+                        filteredTransaction.id.toString(),
+                        groupIndex,
+                        groupData.merkleProof.treeSize,
+                        groupData.merkleProof.leaves.map { it.index },
+                        groupData.merkleProof.hashes.map { it.toString() }
+                    )
+
+                    groupData.merkleProof.leaves.forEach { leaf ->
+
+                        // Persist each merkle proof leaf to the merkle proof leaf table
+                        repository.persistMerkleProofLeaf(
+                            em,
+                            persistedMerkleProofId,
+                            leaf.index
+                        )
+
+                        // Persist the data for each leaf to the component table
+                        repository.persistTransactionComponentLeaf(
+                            em,
+                            filteredTransaction.id.toString(),
+                            groupIndex,
+                            leaf.index,
+                            leaf.leafData,
+                            sandboxDigestService.hash(leaf.leafData, DigestAlgorithmName.SHA2_256).toString()
+                        )
+                    }
+                }
             }
         }
     }
 
-    override fun findMerkleProofs(
-        transactionId: String,
-        groupIndex: Int
-    ): List<MerkleProofDto> {
+    private fun findFilteredTransactions(
+        ids: List<String>
+    ): Map<String, FilteredTransaction> {
```

Comment:
```suggestion
    ): Map<String, List<FilteredTransaction?, List<DigitalSignatureAndMetadata>>> {
```
Did we not agree to persist and fetch signatures alongside a filtered transaction?

Reply:
Yes but I haven't pushed those changes yet 😄